Configure dfclient/dfdaemon/dfget log file path via 'logConfig.path' property

YanzheL opened the pull request:
Signed-off-by: YanzheL lee.yanzhe@yanzhe.org
<!-- 
Please make sure you have read and understood the contributing guidelines;
https://github.com/dragonflyoss/dragonfly/blob/master/CONTRIBUTING.md -->

### Ⅰ. Describe what this PR did

Currently, there is no option for user to customize the log file path, which causes inconvenience while using `dfclient` in container.

For a typical use case, it is expected that the dfclient container logs to stdout so we can inspect the logs via `docker logs -f <dfclient>` instead of checking the log file in default location `~/.small-dragonfly/logs/dfdaemon.log`

This PR enables user to configure the log file path via configuration property `logConfig.path`.

For example
```yaml
logConfig:
  path: /dev/stdout
```

### Ⅱ. Does this pull request fix one issue?
<!--If that, add "fixes #xxxx" below in the next line, for example, fixes #15. Otherwise, add "NONE" -->
NONE

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)

Already tested in container.

### Ⅳ. Describe how to verify it

1. Build the dockerfile

2. Add this property in dfclient configuration file

   ```yaml
   logConfig:
     path: /dev/stdout
   ```

3. You will see the log in `stdout` instead of the default location `~/.small-dragonfly/logs/dfdaemon.log`

### Ⅴ. Special notes for reviews

Starnop commented on pkg/dflog/log.go:
```diff
@@ -72,7 +77,7 @@ func getLumberjack(l *logrus.Logger) *lumberjack.Logger {
 	return func(l *logrus.Logger) error {
 		if f == "" {
-			return nil
+			f = filepath.Join("logs", "dfdaemon.log")
```
In fact, not only dfdaemon will use this function, so do with `dfget` and `supenrode`.

YanzheL replied:
Ok, I rollbacked the change of this line, and set the default log path for dfget/dfdaemon in caller (See my new commits).

As for supernode, I didn't touch its code. Because it [writes to two different log files](https://github.com/dragonflyoss/Dragonfly/blob/def2b217d4faab07dd9fa03e527d3dc3cc106b8a/cmd/supernode/app/root.go#L84-L92), a single `logConfig.path` property is not enough to configure the location of these files.

For now, this PR configures the log path of dfget/dfdaemon, not for supernode.

lowzj commented:
@YanzheL thanks for your contributing very much! Could you please rebase the code and fix the conflict and reviewed comment, then I will merge this pr.

YanzheL commented:
> @YanzheL thanks for your contributing very much! Could you please rebase the code and fix the conflict and reviewed comment, then I will merge this pr.

I updated and rebased the PR, thanks.
